lib/gitarro/backend.rb

```diff
@@ -245,43 +258,37 @@ def failed_status?(comm_st)
 
   # control if the pr change add any files, specified
   # it can be also a dir
-  def empty_files_changed_by_pr
-    return if pr_files.any?
+  def empty_files_changed_by_pr(pr)
+    return if pr_all_files_type(pr.number, @file_type).any?
     puts "no files of type #{@file_type} found! skipping"
     true
   end
 
-  def do_changelog_test(repo, pr)
+  def do_changelog_test(pr)
     @j_status = 'failure'
-    pr_all_files_type(repo, pr.number, @file_type)
     # if the pr contains changes on .changes file, test ok
-    @j_status = 'success' if @pr_files.any?
-    magic_comment(repo, pr.number)
-    @client.create_status(repo, pr.head.sha, @j_status,
-                          context: @context, description: @description,
-                          target_url: @target_url)
-    true
+    @j_status = 'success' if pr_all_files_type(pr.number, @file_type).any?
+    magic_comment(pr.number)
+    create_status(pr, @j_status)
   end
 
   # do the changelog test and set status
-  def changelog_changed(repo, pr, comm_st)
+  def changelog_changed(pr, comm_st)
     return false unless @changelog_test
     # only execute 1 time, don"t run if test is failed, or ok
-    return false if failed_status?(comm_st)
-    return false if success_status?(comm_st)
-    do_changelog_test(repo, pr)
+    return false if failed_status?(comm_st) || success_status?(comm_st)
+    do_changelog_test(pr)
   end
 
   def retrigger_needed?(pr)
     # we want redo sometimes tests
-    return false unless magicword(@repo, pr.number, @context)
+    return false unless magicword(pr.number, @context)
     # changelog trigger
     if @changelog_test
-      do_changelog_test(@repo, pr)
+      do_changelog_test(pr)
       return false
     end
-    pr_all_files_type(@repo, pr.number, @file_type)
-    return false unless @pr_files.any?
+    return false unless pr_all_files_type(pr.number, @file_type).any?
```

Comment:
nitpick:  imho we can refactor this;

```console
return pr_all_files_type(pr.number, @file_type).any?
```
and we can remove line 294 true

Reply:
I think we can just refactor as:

```pr_all_files_type(pr.number, @file_type).any?```

Can't we?

Reply:
Done.